### ISSUE TITLE
ReaR mkrescue stalls in Fedora 30 VM with no error

### ISSUE THREAD

**Issue by adatum** [rear/rear#2203]
#### ReaR version ("/usr/sbin/rear -V"):
Relax-and-Recover 2.4 / 2018-06-21

#### OS version ("cat /etc/rear/os.conf" or "lsb_release -a" or "cat /etc/os-release"):
Fedora 30 Workstation

#### Hardware (PC or PowerNV BareMetal or ARM) or virtual machine (KVM guest or PoverVM LPAR):
VirtualBox 6.0 VM with Fedora 30 guest and Fedora 30 MATE host.

#### System architecture (x86 compatible or PPC64/PPC64LE or what exact ARM device):
x86_64

#### Firmware (BIOS or UEFI or Open Firmware) and bootloader (GRUB or ELILO or Petitboot):
UEFI

#### Description of the issue (ideally so that others can reproduce it):

Running `rear -v mkrescue` starts running then stalls. There is no error, so I don't know what to look for. It does not terminate on its own. I have left it running for hours and nothing changes. `top` shows the VM/guest is idle.

This is all that is seen in the terminal:
```
# rear -v mkrescue
Relax-and-Recover 2.4 / 2018-06-21
Using log file: /var/log/rear/rear-localhost.log

```

The log file:
```
$ cat /var/log/rear/rear-localhost.log
2019-08-10 14:46:54.258707036 Relax-and-Recover 2.4 / 2018-06-21
2019-08-10 14:46:54.259945317 Command line options: /usr/sbin/rear -v mkrescue
2019-08-10 14:46:54.260801937 Using log file: /var/log/rear/rear-localhost.log
2019-08-10 14:46:54.263063373 Including /etc/rear/os.conf
2019-08-10 14:46:54.266459394 Including conf/Linux-i386.conf
2019-08-10 14:46:54.268400033 Including conf/GNU/Linux.conf
2019-08-10 14:46:54.293311513 Including /etc/rear/local.conf
2019-08-10 14:46:54.295457361 ======================
2019-08-10 14:46:54.296540328 Running 'init' stage
2019-08-10 14:46:54.297676346 ======================
2019-08-10 14:46:54.305016885 Including init/default/005_verify_os_conf.sh
2019-08-10 14:46:54.308793055 Including init/default/010_set_drlm_env.sh
2019-08-10 14:46:54.312161073 Including init/default/030_update_recovery_system.sh
2019-08-10 14:46:54.315468506 Including init/default/050_check_rear_recover_mode.sh
2019-08-10 14:46:54.316609203 Finished running 'init' stage in 0 seconds
2019-08-10 14:46:54.325484744 Using build area '/tmp/rear.ykgxnnO1p8aRXXF'
mkdir: created directory '/tmp/rear.ykgxnnO1p8aRXXF/rootfs'
mkdir: created directory '/tmp/rear.ykgxnnO1p8aRXXF/tmp'
2019-08-10 14:46:54.330229936 Running mkrescue workflow
2019-08-10 14:46:54.332002367 ======================
2019-08-10 14:46:54.333464712 Running 'prep' stage
2019-08-10 14:46:54.334683316 ======================
2019-08-10 14:46:54.341591828 Including prep/default/005_remove_workflow_conf.sh
mkdir: created directory '/tmp/rear.ykgxnnO1p8aRXXF/rootfs/etc'
mkdir: created directory '/tmp/rear.ykgxnnO1p8aRXXF/rootfs/etc/rear'
2019-08-10 14:46:54.347427912 Including prep/default/020_translate_url.sh
2019-08-10 14:46:54.352192932 Including prep/default/030_translate_tape.sh
2019-08-10 14:46:54.356191772 Including prep/default/040_check_backup_and_output_scheme.sh
2019-08-10 14:46:54.360378509 Including prep/default/050_check_keep_old_output_copy_var.sh
2019-08-10 14:46:54.364931598 Including prep/default/100_init_workflow_conf.sh
2019-08-10 14:46:54.370318102 Including prep/GNU/Linux/200_include_getty.sh
2019-08-10 14:46:54.389302706 Including prep/GNU/Linux/200_include_serial_console.sh
/usr/share/rear/lib/_input-output-functions.sh: line 328: type: getty: not found
2019-08-10 14:46:54.417695710 Including prep/GNU/Linux/210_include_dhclient.sh
2019-08-10 14:46:54.427675538 Running DHCP client found, enabling USE_DHCLIENT
/usr/share/rear/lib/_input-output-functions.sh: line 328: type: dhcpcd: not found
/usr/share/rear/lib/_input-output-functions.sh: line 328: type: dhcp6c: not found
/usr/share/rear/lib/_input-output-functions.sh: line 328: type: dhclient6: not found
2019-08-10 14:46:54.435084938 Including prep/GNU/Linux/220_include_lvm_tools.sh
2019-08-10 14:46:54.437112752 Device mapper found enabled. Including LVM tools.
2019-08-10 14:46:54.507558907 Including prep/GNU/Linux/230_include_md_tools.sh
2019-08-10 14:46:54.512452559 Including prep/GNU/Linux/240_include_multipath_tools.sh
2019-08-10 14:46:54.515937488 Including prep/GNU/Linux/280_include_systemd.sh
2019-08-10 14:46:54.526689798 Including systemd (init replacement) tool-set to bootstrap Relax-and-Recover
2019-08-10 14:46:54.530972685 Including prep/GNU/Linux/280_include_virtualbox.sh
2019-08-10 14:46:54.536379668 Adding virtualbox modules
2019-08-10 14:46:54.540386814 Including prep/GNU/Linux/280_include_vmware_tools.sh
2019-08-10 14:46:54.546537173 Including prep/GNU/Linux/290_include_drbd.sh
2019-08-10 14:46:54.553692863 Including prep/GNU/Linux/300_check_backup_and_output_url.sh
2019-08-10 14:46:54.558063968 Including prep/ISO/default/300_check_iso_dir.sh
2019-08-10 14:46:54.561495826 Including prep/GNU/Linux/300_include_grub_tools.sh
/usr/share/rear/lib/_input-output-functions.sh: line 328: type: grub-probe: not found
2019-08-10 14:46:54.570428115 Including prep/GNU/Linux/310_include_cap_utils.sh
2019-08-10 14:46:54.574747902 Including prep/ISO/default/320_check_cdrom_size.sh
2019-08-10 14:46:54.577392302 ISO Directory '/var/lib/rear/output' [/dev/mapper/fedora_localhost--live-root] has 8667 MB free space
2019-08-10 14:46:54.581175505 Including prep/default/320_include_uefi_env.sh
```

With debug (`rear -vD mkrescue`), the part of the log looks like this:

```
2019-08-10 19:24:32.213442574 Including prep/default/320_include_uefi_env.sh
2019-08-10 19:24:32.214930267 Entering debugscripts mode via 'set -x'.
+ source /usr/share/rear/prep/default/320_include_uefi_env.sh
++ grep -qw noefi /proc/cmdline
++ is_false
++ case "$1" in
++ return 1
++ modprobe -q efivars
++ SYSFS_DIR_EFI_VARS=
++ [[ -d /sys/firmware/efi/vars ]]
++ [[ -d /sys/firmware/efi/efivars ]]
++ SYSFS_DIR_EFI_VARS=/sys/firmware/efi/efivars
++ grep -qw efivars /proc/mounts
++ SYSFS_DIR_EFI_VARS=/sys/firmware/efi/efivars
+++ find /boot -maxdepth 1 -iname efi -type d
++ test /boot/efi
++ local esp_mount_point=
++ esp_mount_point='/\/boot\/efi/'
+++ awk /proc/mounts

```

**Comment by adatum** [comment]
The problem went away after updating and rebooting the guest OS. 

I had not wanted to download and install ~1GB of updates for the guest, as it is a throwaway VM. Originally it was on kernel 5.0.x and now it's 5.2.6.  

I have no idea what the issue was but now `rear mkrescue` completes successfully. Closing issue.

**Comment by jsmeix** [comment]
@adatum
it seems this one is a duplicate of
https://github.com/rear/rear/issues/2095
which is fixed in ReaR 2.5 - look for 'awk' in
http://relax-and-recover.org/documentation/release-notes-2-5
 
Better use ReaR 2.5 because that is the only version
where we at ReaR upstream will fix bugs.
Alternatively if you need backports of fixes you should get
in contact with your Linux distributor.
